geode-core/src/main/java/org/apache/geode/internal/security/shiro/JMXShiroAuthenticator.java

```diff
@@ -48,13 +48,16 @@ public JMXShiroAuthenticator(SecurityService securityService) {
 
   @Override
   public Subject authenticate(Object credentials) {
-    String username = null;
+    String username;
     Properties credProps = new Properties();
     if (credentials instanceof String[]) {
       final String[] aCredentials = (String[]) credentials;
       username = aCredentials[0];
       credProps.setProperty(ResourceConstants.USER_NAME, aCredentials[0]);
       credProps.setProperty(ResourceConstants.PASSWORD, aCredentials[1]);
+    } else if (credentials instanceof String) {
```

Comment:
did we decide to do something about this now or later?

Reply:
I think we should do this later when we need it. It is not clear that we will ever need a more general container (for example a Map) of these properties.